### PR TITLE
Document how to use dcos-monitoring with DSS

### PR DESCRIPTION
**Pull request by branden**
## Description

https://jira.mesosphere.com/browse/DCOS-51042

This adds documentation explaining how to deploy dcos-monitoring's provided Prometheus, Grafana, and Alertmanager with volumes provided by DSS. https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/

This adds a DSS Volume section to the Storage page for Prometheus, and adds new Storage pages for Grafana and Alertmanager with DSS Volume sections.

## Urgency
- [ ] Blocker 
- [ ] High
- [x] Medium

## Requirements
- Test all commands and procedures.
- Add [redirects](https://github.com/mesosphere/dcos-docs-site/wiki/Redirects).
- Change all affected versions (e.g. 1.7, 1.8, 1.9, 1.10, 1.11, 1.12).
- See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing).

### CODE REVIEW

File: pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+## DSS Volume
+
+You can configure Alert Manager to use a dedicated volume provided by the [DC/OS Storage Service](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/) (DSS).
+
+At least one volume must be available for Alert Manager to use. For more on creating volumes, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-volume/dcos-storage-volume-create).
+
+When configuring Alert Manager, select `MOUNT` for its volume `type`, choose the minimum `size` for the volume in MB, and enter the `profile` for the volume. For more on volume profiles, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-profile/).
+
+The following configuration will deploy Alert Manager with a `fast` DSS volume that is 512 MB or larger. If there is no such volume available, Alert Manager will fail to deploy.
```
Comment (masonse) [inline pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md]: "The following configuration will deploy Alert Manager with a `fast` DSS volume that is 512 MB or larger."  change to: "The following configuration will deploy Alert Manager to a volume that has a volume profile of `fast` which is 512MB or larger.

File: pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#include /services/include/beta-software-warning.tmpl
+
+# Configuring Alert Manager Storage
+
+## DSS Volume
+
+You can configure Alert Manager to use a dedicated volume provided by the [DC/OS Storage Service](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/) (DSS).
+
+At least one volume must be available for Alert Manager to use. For more on creating volumes, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-volume/dcos-storage-volume-create).
```
Comment (masonse) [inline pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md]: Is there any minimal requirements for this volume that we should recommend?  Should Alert Manager volumes be on high performance storage (e.g. SSD backed)?  Is the example size below (512MB) sufficient?  Or should it be larger.  I would definitely include a recommendation for storage guidelines.

File: pages/services/beta-dcos-monitoring/0.4.3-beta/operations/grafana/storage/index.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## DSS Volume
+
+You can configure Grafana to use a dedicated volume provided by the [DC/OS Storage Service](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/) (DSS).
+
+At least one volume must be available for Grafana to use. For more on creating volumes, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-volume/dcos-storage-volume-create).
+
+When configuring Grafana, select `MOUNT` for its volume `type`, choose the minimum `size` for the volume in MB, and enter the `profile` for the volume. For more on volume profiles, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-profile/).
+
+The following configuration will deploy Grafana with a `fast` DSS volume that is 512 MB or larger. If there is no such volume available, Grafana will fail to deploy.
```
Comment (masonse) [inline pages/services/beta-dcos-monitoring/0.4.3-beta/operations/grafana/storage/index.md]: "The following configuration will deploy Grafana with a fast DSS volume that is 512 MB or larger." change to: "The following configuration will deploy Grafana to a volume that has a volume profile of fast which is 512MB or larger."

File: pages/services/beta-dcos-monitoring/0.4.3-beta/operations/grafana/storage/index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+#include /services/include/beta-software-warning.tmpl
+
+# Configuring Grafana Storage
+
+## DSS Volume
+
+You can configure Grafana to use a dedicated volume provided by the [DC/OS Storage Service](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/) (DSS).
+
+At least one volume must be available for Grafana to use. For more on creating volumes, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-volume/dcos-storage-volume-create).
```
Comment (masonse) [inline pages/services/beta-dcos-monitoring/0.4.3-beta/operations/grafana/storage/index.md]: Is there any minimal requirements for this volume that we should recommend? Should Grafana volumes be on high performance storage (e.g. SSD backed)? Is the example size below (512MB) sufficient? Or should it be larger. I would definitely include a recommendation for storage guidelines. I would also align the example to your guidelines.

File: pages/services/beta-dcos-monitoring/0.4.3-beta/operations/prometheus/storage/index.md [hunk 5]
```diff
@@ -26,3 +26,27 @@ You can configure the storage retention (default: 15d) of Prometheus by setting
...
+
+## DSS Volume
+
+You can configure Prometheus to use a dedicated volume provided by the [DC/OS Storage Service](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/) (DSS).
+
+At least one volume must be available for Prometheus to use. For more on creating volumes, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-volume/dcos-storage-volume-create).
+
+When configuring Prometheus, select `MOUNT` for its volume `type`, choose the minimum `size` for the volume in MB, and enter the `profile` for the volume. For more on volume profiles, see the [DSS documentation](https://docs.mesosphere.com/services/beta-storage/0.5.3-beta/cli-references/dcos-storage-profile/).
+
+The following configuration will deploy Prometheus with a `fast` DSS volume that is 25 GB or larger. If there is no such volume available, Prometheus will fail to deploy.
```
Comment (masonse) [inline pages/services/beta-dcos-monitoring/0.4.3-beta/operations/prometheus/storage/index.md]: Is there any minimal requirements for this volume that we should recommend? Should Prometheus volumes be on high performance storage (e.g. SSD backed)? Is the example size below (512MB) sufficient? Or should it be larger. I would definitely include a recommendation for storage guidelines. I would also align the example to your guidelines.

File: pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md (hunk 2 above)
Comment (branden) [reply pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md]: I used the default size values from dcos-monitoring's configuration. I've deployed dcos-monitoring a bunch with these storage sizes, so it should be OK.

Comment (branden) [reply pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md]: I don't know whether Alert Manager specifically needs fast storage. I think explaining how to decide whether it needs a certain type of storage is out of scope of this doc.

Comment (masonse) [reply pages/services/beta-dcos-monitoring/0.4.3-beta/operations/alertmanager/storage/index.md]: If it doesn't need fast storage then a recommendation isn't needed.

**Comment by branden** [comment]
Closing in favor of https://github.com/mesosphere/dcos-monitoring/pull/145.

**Pull request closed without merging**